### PR TITLE
[chore] [mdatagen] Mdatagen hack metadata

### PR DESCRIPTION
**Pull request by BominRahmani**
**Description:** Changed what mdatagen outputs into the generated_status_go file for receivers. If the receiver has the suffix "receiver", it will get rid of that. This will allow us to also get rid of the "typeStr" variable for most of the receivers and standardize it into metadata.Type. The same way we have been doing for stability -> metadata.Stability

Within this change is also the make generate update for generated_status_go files for the receivers that have had that bit added in.

### CODE REVIEW

**Comment by BominRahmani** [comment]
This is the first time i have messed with mdatagen and I know it is connected to a lot of things. I ran make gotest and everything seemed to be fine. If this change has broader implications that I have not thought about or foreseen, please let me know. Otherwise I think this should be fine!

**Comment by dmitryax** [comment]
Is this intended to be a temporary solution until https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21382 is resolved?

**Comment by BominRahmani** [comment]
@dmitryax This was intended to be a solution, but if someone else had another way of doing it thats cleaner or adds more functionality in the long run, then this could be a temporary. I just wanted to try to get this in so when I go through and and do <https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21213>, I can also handle the changing of the typeStr on the receivers.

File: receiver/nginxreceiver/README.md
Comment (atoulme) [inline receiver/nginxreceiver/README.md:4]: not sure this change is related?

File: receiver/activedirectorydsreceiver/internal/metadata/generated_status.go
Comment (atoulme) [inline receiver/activedirectorydsreceiver/internal/metadata/generated_status.go:10]: The type registered in factory.go is `active_directory_ds`. Unfortunately, this change is not enough to accomodate.

Comment (BominRahmani) [reply receiver/activedirectorydsreceiver/internal/metadata/generated_status.go:10]: Ah, you are right. This one is a bit strange with the underscores. Do you have any solutions to this in mind, I can take a look into implementing a fix if you have an idea.

Comment (atoulme) [reply receiver/activedirectorydsreceiver/internal/metadata/generated_status.go:10]: No, for now, I lean towards taking your fix, and we will need to iterate one more time for this component and a few others.

**Comment by atoulme** [comment]
Please rebase off main to pick up the fix when checking licenses.

File: receiver/activedirectorydsreceiver/internal/metadata/generated_status.go
Comment (dmitryax) [reply receiver/activedirectorydsreceiver/internal/metadata/generated_status.go:10]: Another solution: align type in metadata.yaml with what we need, like `active_directory_ds` in this case. And use the directory name for Scope.Name. This solution will work even with https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21469

Comment (dmitryax) [reply receiver/activedirectorydsreceiver/internal/metadata/generated_status.go:10]: No, we should keep the type as is. Otherwise, it'll be a significant breaking change. I mean, we set `active_directory_ds` in metadata.yaml to have it rendered here. But the https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21469 can be solved by setting the directory in the Scope.Name field instead of this type

**Review by atoulme — APPROVED** [review]

**Comment by dmitryax** [comment]
Why do we need to merge this PR if the generated type cannot be used yet?

**Comment by atoulme** [comment]
I think I can make it work for this PR: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21275

**Comment by dmitryax** [comment]
#21275 already have this part on the query/validation path https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21275/files#diff-35261d8aa538a618698be27fbb71cd350cdf5db0f9693652ceaa151b26a78409R57 ? :)

File: receiver/activedirectorydsreceiver/internal/metadata/generated_status.go
Comment (dmitryax) [reply receiver/activedirectorydsreceiver/internal/metadata/generated_status.go:10]: I put a wrong link in my comments. I wanted to mention a solution for https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21382

**Comment by atoulme** [comment]
True, I can make it work either way, I could drop that code though if this one comes in.

**Comment by BominRahmani** [comment]
So what is the consensus on this? Do you guys want me to start looking at @dmitryax solution for this and drop this PR? I'd really like to get this or the other solution in ASAP so I can start knocking out the <https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21213>.

**Comment by dmitryax** [comment]
I don't see how this PR can resolve https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21382 and unblock https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21213

**Comment by dmitryax** [comment]
I submitted https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/21501. Once it's merged, you can set the actual component type in the metadata.yaml

**Comment by djaglowski** [comment]
Superseded by https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/21382

**Pull request closed without merging**
